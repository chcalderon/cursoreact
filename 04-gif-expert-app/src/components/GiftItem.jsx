import PropTypes from 'prop-types';

export const GiftItem = ({title,url,id}) => {
 
  return (
    <div className="card">
      <img src={url} alt={title} />
      <a>{title}</a>
    </div>
  )
}

GiftItem.propTypes = {
  title:PropTypes.string.isRequired,
  url  :PropTypes.string.isRequired
}