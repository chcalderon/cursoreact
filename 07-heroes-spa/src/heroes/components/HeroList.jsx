import { HeroCard } from './';
import { getHeroesByPublisher } from '../helpers';

export const HeroList = ( {publisher} ) => {
  
  const heroes = getHeroesByPublisher(publisher);

  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3">
        {/*!heroes ? (<li>No hay heroes</li>):*/
        heroes.map( hero => (
            <HeroCard 
                key={hero.id} 
                {...hero}
            />
            // <li key={hero.id}>
            //     {hero.superhero}
            // </li>
        ))
            
        }
        
    </div>
  )
}
