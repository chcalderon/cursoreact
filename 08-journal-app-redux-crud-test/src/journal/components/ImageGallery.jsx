import { ImageList, ImageListItem } from "@mui/material";

function srcset(image, size, rows = 1, cols = 1) {
  return {
    src: `${image}?w=${size * cols}&h=${size * rows}&fit=crop&auto=format`,
    srcSet: `${image}?w=${size * cols}&h=${
      size * rows
    }&fit=crop&auto=format&dpr=2 2x`,
  };
}

export const ImageGallery = ({ images }) => {
  return (
    <ImageList
      sx={{ width: '100%', height: 450 }}
      variant="quilted"
      cols={4}
      rowHeight={250}
    >
      {images.map((image) => (
        <ImageListItem key={image} cols={ 1} rows={ 1}>
          <img
            {...srcset(image, 121, 1, 1)}
            alt="Imagen de la nota"
            loading="lazy"
          />
        </ImageListItem>
      ))}
    </ImageList>
  );
}
