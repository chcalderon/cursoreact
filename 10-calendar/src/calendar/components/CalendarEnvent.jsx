
export const CalendarEnvent = ({ event }) => {
    // console.log(props);
    const { title, user } = event;
  return (
    <>
      <strong>{title}</strong>
      <strong> - {user.name}</strong>
    </>
  )
}
