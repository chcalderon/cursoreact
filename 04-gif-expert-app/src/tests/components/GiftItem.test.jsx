import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../components";

describe('Pruebas en <GiftItem/>', () => { 
    
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('debe de hacer match con el snapshot', () => { 
        
        const {container}=render( <GiftItem title={title} url={url} /> );
        expect(container).toMatchSnapshot();

     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GiftItem title={title} url={url} /> );
        //screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
      });

    test('debe de mostrar el titulo en el componente', () => { 
        
        render( <GiftItem title={title} url={url} /> );
        expect( screen.getByText(title)).toBeTruthy();
     })
 });