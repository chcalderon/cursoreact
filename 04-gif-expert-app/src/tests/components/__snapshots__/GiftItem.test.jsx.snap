// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <a>
      Saitama
    </a>
  </div>
</div>
`;
